dotnet/src/AutoGen.Core/Agent/MiddlewareStreamingAgent.cs

```diff
@@ -42,7 +35,20 @@ public class MiddlewareStreamingAgent : MiddlewareAgent, IMiddlewareStreamAgent
     /// </summary>
     public IEnumerable<IStreamingMiddleware> StreamingMiddlewares => _streamingMiddlewares;
 
-    public Task<IAsyncEnumerable<IStreamingMessage>> GenerateStreamingReplyAsync(IEnumerable<IMessage> messages, GenerateReplyOptions? options = null, CancellationToken cancellationToken = default)
+    public string Name { get; }
+
+    public Task<IMessage> GenerateReplyAsync(IEnumerable<IMessage> messages, GenerateReplyOptions? options = null, CancellationToken cancellationToken = default)
+    {
+        IStreamingAgent agent = this._agent;
+        foreach (var middleware in _streamingMiddlewares)
+        {
+            agent = new DelegateStreamingAgent(middleware, agent);
+        }
```

Comment:
Seems like every time we `GenerateReplyAsync` we will iterate through `_streamingMiddlewares` and assign them. 

Is it better to do this work once `new DelegateStreamingAgent(middleware, agent);` in the constructor and in `UseStreaming()` ?

Reply:
Thanks will address it in later commit